jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -13,8 +13,357 @@
+import org.eclipse.jetty.server.ResourceService;
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.URIUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class DefaultServlet extends HttpServlet
 {
+    private ResourceService _resourceService;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException
+    {
+        ContextHandler contextHandler = initContextHandler(config.getServletContext());
+
+        _resourceService = new ServletResourceService();
+        MimeTypes mimeTypes = new MimeTypes();
+        CompressedContentFormat[] precompressedFormats = new CompressedContentFormat[0];
+        _resourceService.setContentFactory(new CachingContentFactory(new ResourceContentFactory(contextHandler.getResourceBase(), mimeTypes, precompressedFormats)));
+
+        // TODO init other settings
+    }
+
+    protected ContextHandler initContextHandler(ServletContext servletContext)
+    {
+        ContextHandler.Context context = ContextHandler.getCurrentContext();
+        if (context == null)
+        {
+            if (servletContext instanceof ContextHandler.Context)
+                return ((ContextHandler.Context)servletContext).getContextHandler();
+            else
+                throw new IllegalArgumentException("The servletContext " + servletContext + " " +
+                    servletContext.getClass().getName() + " is not " + ContextHandler.Context.class.getName());
+        }
+        else
+            return context.getContextHandler();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
+    {
+        boolean useOutputDirectByteBuffers = true;
+        if (resp instanceof ServletContextResponse.ServletApiResponse servletApiResponse)
+            useOutputDirectByteBuffers = servletApiResponse.getResponse().getWrapped().getRequest().getConnectionMetaData().getHttpConfiguration().isUseOutputDirectByteBuffers();
+
+        HttpContent content = _resourceService.getContent(req.getServletPath(), resp.getBufferSize());
+        if (content == null)
+        {
+            // no content
+            resp.setStatus(404);
+        }
+        else
+        {
+            // serve content
+            try
+            {
+                _resourceService.doGet(new ServletGenericRequest(req), new ServletGenericResponse(resp, useOutputDirectByteBuffers), Callback.NOOP, content);
+            }
+            catch (Exception e)
+            {
+                throw new ServletException(e);
+            }
+        }
+    }
+
+    @Override
+    protected void doHead(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
+    {
+        doGet(req, resp);
+    }
+
+    private static class ServletGenericRequest implements ResourceService.GenericRequest
```

Comment:
This class should be a wrapper of the `HttpServletRequest` to a core `Request` as a top level class, that will have other uses.